MetaMask/metamask-extension
# Replace `eth-optimism/contracts` package with local snippet of gas price ABI and address

Fixes: #16565
See also: https://consensys.slack.com/archives/C01KXHPU3C6/p1662579353556779 (looks like all optimism team messages have been deleted though lol)

## Explanation
 We don't need to pull in  `@eth-optimism/contracts` with all the additional code and dependencies that it brings. We can do the query to the optimism gas price oracle with a snippet of the oracle contract ABI and its address -> instantiate the contract and call to it ourselves to get the L1 gas fee estimate.
 
 This PR simply removes the use of `@eth-optimism/contracts` helper method to achieve this and implements the required call all within our own `fetchEstimatedL1Fee` helper function.

To test:
- Attempt to initiate a transaction on Optimism
- Go to devtools -> source -> `fetchEstimatedL1Fee.js`
- Put a break point on line 41 (`return result?.toHexString();`)
- Check that the result returned is defined and is typed BigNumber. This indicates that we're successfully calling to the Optimism Gas price oracle.

## Pre-merge author checklist
